### DISCUSSION TITLE
Global Properties in Markdown or Mustaching Markdown 🥸

### DISCUSSION THREAD

**Discussion by VaclavElias** [dotnet/docfx#8611] [Ideas]
I have got this looong path in many places in our `*.md` files (these are not templates!) and I don't fancy that path being hard-coded.

```
## Code

[!code-csharp[Entity](../../../../stride/samples/Tutorials/CSharpBeginner/CSharpBeginner/CSharpBeginner.Game/Code/GettingTheEntityDemo.cs)]
```

Maybe I would like this version with `_strideLocalPath` (remote GitHub link would be also nice in some cases)

```
## My lovely path

{{_strideLocalPath}}

{{#_names}}
  	<p>Name: {{.}}</p>
{{/_names}}

## Code

[!code-csharp[Entity]({{_strideLocalPath}}/GettingTheEntityDemo.cs)]
```

The property is defined in `docfx.json`

```json
    "globalMetadata": {
      "_appTitle": "Stride Docs",
      "_enableSearch": true,
      "_appLogoPath": "media/stride-logo-red.svg",
      "_strideLocalPath": "../../../../stride/samples/Tutorials/CSharpBeginner/CSharpBeginner/CSharpBeginner.Game/Code",
      "_names": [ "vicancy", "yufeih" ]
    },
```
And this is the result, including other mustache goodies, like a loop 🙂

![image](https://user-images.githubusercontent.com/4528464/230789551-2ef7156b-47b8-4c1a-825e-a89a43933ea3.png)

At the moment it is not possible, unless 🤩 I add this code to `BuildConceptualDocument.cs`.

⚠️Note: This is just my test implementation. You are going to know the best how it should be done..

Not sure, if this is the right place, also the consequences. Maybe it would be better to add opt-in option for Mustache preprocessing in the file `.md` yaml header.

```csharp
var stubble = new StubbleBuilder().Build();
var mustachedMarkdown = stubble.Render(markdown, content);

if (markdown != mustachedMarkdown)
{
    Logger.LogWarning($"Mustache applied for {model.File}");

    markdown = mustachedMarkdown;
}
```

![image](https://user-images.githubusercontent.com/4528464/230789809-581e4f95-41a8-4b18-a34c-8858f7305682.png)

What are you thoughts on this?

**Comment by yufeih** [comment]
The [`codeSourceBasePath`](https://github.com/dotnet/docfx/blob/main/src/Microsoft.DocAsCode.Dotnet/MetadataJsonConfig.cs#L72) config sets the base directory for the path specified in `<code source="Example.cs">`

**Reply by VaclavElias** [reply]
If I used `codeSourceBasePath`, then I don't have to use the long path in this file? I think you might be referring to use `codeSourceBasePath` in some other scenarios?

https://github.com/VaclavElias/stride-docs-next/blob/main/en/tutorials/csharpbeginner/delta-time.md?plain=1

![image](https://user-images.githubusercontent.com/4528464/230963273-f577aeea-0a43-4855-8ac2-e5eaeaeb0fc5.png)
